app/controllers/api/base_controller/parser.rb

```diff
@@ -331,6 +323,42 @@ def assert_all_required_fields_exists(data, type, required_fields)
           raise BadRequestError, "Resource #{missing_fields.join(", ")} needs be specified for creating a new #{type}"
         end
       end
+
+      def validate_provider_class
+        param = params['provider_class']
+        return unless param.present?
+
+        raise BadRequestError, "Unsupported provider_class #{param} specified" if param != "provider"
+        %w(tags policies policy_profiles).each do |cname|
+          if @req.subcollection == cname || @req.expand?(cname)
+            raise BadRequestError, "Management of #{cname} is unsupported for the Provider class"
+          end
+        end
+        @collection_klasses[:providers] = Provider
+      end
+
+      def validate_collection_class
+        param = params['collection_class']
+        return unless param.present?
+
+        begin
+          param_klass = param.constantize
+        rescue
+          raise BadRequestError, "Invalid collection_class #{param} specified"
+        end
+
+        collection_klass = collection_config[@req.collection].klass.constantize
+
+        # If it is the collection's class, then we're good to go as is.
+        return if param_klass == collection_klass
+
+        if param_klass < collection_klass
```

Comment:
I believe you can do `<=` and just collapse this conditional with the one above.